.github/workflows/build.yml

```diff
@@ -28,6 +28,82 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  prebuild-legion:
+    name: Prebuild Legion with CMake
+    runs-on: ubuntu-20.04
+    defaults:
+      run:
+        shell: bash -l {0} # required to use an activated conda environment
+    strategy:
+      matrix:
+        gpu_backend: ["cuda", "hip_rocm"]
+      fail-fast: false
+    steps:
+      - name: Checkout Git Repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Free additional space on runner
+        run: .github/workflows/helpers/free_space_on_runner.sh
+
+      - name: Install CUDA
+        uses: Jimver/cuda-toolkit@v0.2.11
+        id: cuda-toolkit
+        with:
+          cuda: "11.8.0"
+          # Disable caching of the CUDA binaries, since it does not give us any significant performance improvement
+          use-github-cache: "false"
+
+      - name: Install system dependencies
+        run: FF_GPU_BACKEND=${{ matrix.gpu_backend }} .github/workflows/helpers/install_dependencies.sh
+
+      - name: Install conda and FlexFlow dependencies
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          activate-environment: flexflow
+          environment-file: conda/environment.yml
+          auto-activate-base: false
+        
+      - name: Build Legion
+        env: 
+          FF_GPU_BACKEND: ${{ matrix.gpu_backend }}
+        run: |
+          export PATH=/opt/conda/bin:$PATH
+          export CUDNN_DIR=/usr/local/cuda
+          export CUDA_DIR=/usr/local/cuda
+          export FF_CUDA_ARCH=all
+
+          export BUILD_LEGION_ONLY=ON
+
+          cores_available=$(nproc --all)
+          n_build_cores=$(( cores_available -1 ))
+
+          mkdir build
+          cd build
+          ../config/config.linux
+          make -j $n_build_cores
+          ../config/config.linux
+          make install
+      
+      - name: Prepare library files 
+        env: 
+          FF_GPU_BACKEND: ${{ matrix.gpu_backend }}
+        run: |
+          echo "Removing unnecessary files..."
```

Comment:
I don't think this is necessary here, since we are not using the `placeholder.cpp` trick from the `flexflow-third-party` repo